modules/loader-utils/src/lib/file-provider/range-request-file.ts

```diff
@@ -0,0 +1,137 @@
+// loaders.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+import {FileProvider} from './file-provider';
```

Comment:
Check differences with HTTPFile (which does inherit from ReadableFil): https://github.com/visgl/loaders.gl/blob/3c5f743bebd14d3e465e9eedb279e5eb1c6f370b/modules/loader-utils/src/lib/files/http-file.ts#L67

We should really have some tests for this. Maybe we can mock fetch?